src/OrchardCore.Modules/OrchardCore.AutoSetup/AutoSetupMiddleware.cs

```diff
@@ -146,123 +134,24 @@ public async Task InvokeAsync(HttpContext httpContext)
                         {
                             if (_setupOptions != setupOptions)
                             {
-                                await CreateTenantSettingsAsync(setupOptions);
+                                await autoSetupService.CreateTenantSettingsAsync(setupOptions);
                             }
                         }
                     }
 
                     httpContext.Response.Redirect(pathBase);
 
+                    return;
+                }
+                else
```

Comment:
This would prevent the error message from being displayed to the user. Like server-side validation or database creation error.

Even if you still decide to keep 503 as the code.

Reply:
@sebastienros : Just committed 
[Won't stop other Middlewares after success](https://github.com/OrchardCMS/OrchardCore/pull/16834/commits/debdf12e1c4c640f1d83d979739a3b84cece0328)